corehq/apps/app_manager/tests/data/suite/suite-case-detail-tabs-with-nodesets.xml

```diff
@@ -144,17 +144,20 @@
           </text>
         </template>
       </field>
-      <field>
-        <header>
-          <text>
-            <locale id="m0.case_long.case_gender_6.header"/>
-          </text>
-        </header>
-        <template>
-          <text>
-            <xpath function="gender"/>
-          </text>
-        </template>
+       <field>
+         <header width="0">
+           <text/>
+         </header>
+         <template width="0">
```

Comment:
So in addition to defaulting to use this for sort it also hides this column completely. That seems like a significant change

Reply:
The "gender" detail column is added as "invisible" in 

https://github.com/dimagi/commcare-hq/blob/502c9cd79de296d6c16f0a83d4e21e7453f16422/corehq/apps/app_manager/tests/data/suite/app_case_detail_tabs_with_nodesets.json#L804

which is used to generate the suite file and then compare with this xml in a test 
https://github.com/dimagi/commcare-hq/blob/8e6cd3521bfd4ee6972dc4bb00e433dd083cdba4/corehq/apps/app_manager/tests/test_suite.py#L500

the feature flag to sort by invisible columns is not set in that test.

It's not clear to me why originally, this file would even have this field added as visible. Confused by two "field" nodes under this `detail` tab here as well.

It makes sense that now this should always be hidden, irrespective of the setting/checkbox and have width set to 0 to be sort only details.

Reply:
@snopoke is right, I didn't realize it until this conversation. Because of the logic in HideShortHeaderColumn/HideShortColumn, prior to this change, if the sorting checkbox was unchecked, Search Only fields on case details would get displayed, same as if they used the Plain format: https://github.com/dimagi/commcare-hq/blob/d5fb059e219a949b61809933203ea0c52c3fc864/corehq/apps/app_manager/detail_screen.py#L310-L329

I don't think that's a reasonable use case - any fields like that should just use the plain format, not search-only - but I should be able to tweak the gist to see if there are any apps that were mis-configured that way.

Reply:
It looks like there are two projects that are configured in this way. I'll follow up on the ticket.

Reply:
Followed up on ticket, confirmed that the projects configured in this way are mis-configured and are okay with this PR.

Reply:
Interesting, so looks like the original implementation got it wrong. Thanks for following up on that.